### PR TITLE
terminal ui: use dcrlibwallet/master

### PR DESCRIPTION
**Pull request by itswisdomagain**
Removes all reliance of the terminal ui interface on the `app` package and uses dcrlibwallet/master directly. Similar work will need to be done for the cli, nuklear and web interfaces in successive PRs.

Also, moves the sync progress indicator to the overview page:
Screenshot 2019-09-26 at 1 30 41 PM

### CODE REVIEW

**Review by Sirmorrison — COMMENTED** [review]
image

Interface doesn't load completely on launch of the terminal interface until a options selected.

File: terminal/pages/overview.go [hunk 1]
```diff
@@ -109,38 +122,189 @@ func renderRecentActivity(overviewPage *tview.Flex, wallet walletcore.Wallet, tv
...
+			SetMaxWidth(3).
+			SetExpansion(1)
+		historyTable.SetCell(nextRowIndex, 2, amountCell)
+
+		status := "Pending"
+		confirmations := commonPageData.wallet.GetBestBlock() - tx.BlockHeight + 1
+		if tx.BlockHeight != -1 && confirmations > dcrlibwallet.DefaultRequiredConfirmations {
+			status = "Confirmed"
 		}
+		statusCell := tview.NewTableCell(fmt.Sprintf("%12s", status)).
```
Comment (Sirmorrison) [inline terminal/pages/overview.go:154]: Line space between 153 and 154.

File: terminal/pages/receive.go [hunk 2]
```diff
@@ -144,11 +145,12 @@ func receivePage(wallet walletcore.Wallet, hintTextView *primitives.TextView, se
 		generateAdressFunc(formButton)
 
 		body.AddItem(formDropdown, 2, 0, true)
-		hintTextView.SetText("TIP: Select Prefered Account and hit ENTER to generate Address, \nMove around with TAB, ESC to return to navigation menu")
+		commonPageData.hintTextView.SetText("TIP: Select PreferredAccount and hit ENTER to generate Address, \n" +
```
Comment (Sirmorrison) [inline terminal/pages/receive.go]: `PreferredAccount` should be `Preferred Account`

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

File: terminal/pages/overview.go (hunk 1 above)
Comment (itswisdomagain) [reply terminal/pages/overview.go:154]: The entire lines 149 to 154 are related, don't see why you want a separation of concerns. They are all related to the status to be displayed on the status cell on the table.

**Review by Sirmorrison — APPROVED** [review]

File: cmd/godcr-terminal/terminal.go [hunk 3]
```diff
@@ -1,155 +1,17 @@
...
-		return nil, fmt.Errorf("\nError checking if wallet has been created with dcrwallet previously.\n%s",
-			walletCheckError.Error())
-	}
-	if !walletExists {
-		return nil, fmt.Errorf("\nWallet has not been created with dcrwallet daemon.")
-	}
-
-	return rpcWalletMiddleware, nil
+func main() {
+	terminal.LaunchUserInterface(appDisplayName, defaultAppDataDir, "testnet3")
```
Comment (Sirmorrison) [inline cmd/godcr-terminal/terminal.go:16]: Why are we hard coding the netType? `testnet3`

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

File: cmd/godcr-terminal/terminal.go (hunk 3 above)
Comment (itswisdomagain) [reply cmd/godcr-terminal/terminal.go:16]: It's a temporary stuff, will fix for all interfaces in a subsequent PR.

File: terminal/pages/overview.go [hunk 4]
```diff
@@ -109,38 +122,189 @@ func renderRecentActivity(overviewPage *tview.Flex, wallet walletcore.Wallet, tv
...
+
+		fmt.Sprintf("Fetched %d of %d block headers.", headersFetchProgress.FetchedHeadersCount,
+			headersFetchProgress.TotalHeadersToFetch),
+
+		fmt.Sprintf("%d%% through step 1 of 3.", headersFetchProgress.HeadersFetchProgress),
+
+		fmt.Sprintf("Your wallet is %s behind.",
+			dcrlibwallet.CalculateDaysBehind(headersFetchProgress.CurrentHeaderTimestamp)),
+	}
+	listener.updateUI(report)
```
Comment (Sirmorrison) [inline terminal/pages/overview.go:226]: line space.

File: terminal/pages/overview.go [hunk 5]
```diff
@@ -109,38 +122,189 @@ func renderRecentActivity(overviewPage *tview.Flex, wallet walletcore.Wallet, tv
...
+	var report = []string{
+		fmt.Sprintf("%d%% completed, %s remaining.", headersRescanProgress.TotalSyncProgress,
+			dcrlibwallet.CalculateTotalTimeRemaining(headersRescanProgress.TotalTimeRemainingSeconds)),
+
+		fmt.Sprintf("Scanning %d of %d block headers.", headersRescanProgress.CurrentRescanHeight,
+			headersRescanProgress.TotalHeadersToScan),
+
+		fmt.Sprintf("%d%% through step 3 of 3.", headersRescanProgress.RescanProgress),
+	}
+	listener.updateUI(report)
```
Comment (Sirmorrison) [inline terminal/pages/overview.go:258]: line space

File: terminal/pages/overview.go [hunk 6]
```diff
@@ -109,38 +122,189 @@ func renderRecentActivity(overviewPage *tview.Flex, wallet walletcore.Wallet, tv
...
+}
+
+func (listener *syncProgressListener) OnSyncCanceled(willRestart bool) {}
+
+func (listener *syncProgressListener) OnSyncEndedWithError(err error) {
+	// remove previous update views and error view
+	for _, view := range listener.updateViews {
+		listener.overviewPage.RemoveItem(view)
+	}
+	listener.overviewPage.RemoveItem(listener.peerCountTextView)
```
Comment (Sirmorrison) [inline terminal/pages/overview.go]: line space.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

**Review by Sirmorrison — APPROVED** [review]
Tested and looks good.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
merge conflict please resolve.

**Review by Sirmorrison — APPROVED** [review]

**Pull request merged**